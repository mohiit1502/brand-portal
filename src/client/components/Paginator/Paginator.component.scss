@import "../../styles/override/variables";

.c-Paginator {
  &.table-manage-row {
    //position: relative;
    .count {
      color: $market-place-blue;
    }

    .count-toggle-btn {
      border: $default-input-border;
      border-radius: $input-border-radius;
      font-weight: bold;
      font-size: 14px;

      &:focus, &:active {
        box-shadow: 0px 3px 18px #7AC4E433;
        border: 1px solid #7AC4E4;
        outline: 0;
      }
    }

    .count-dropdown-menu {
      box-shadow: 0 2px 18px 0 rgba(0, 0, 18, 0.15);

      .dropdown-item {
        cursor: pointer;

        &:hover {
          background: $dropdown-hover-background;
        }

        &:active {
          background: $dropdown-active-background;
        }
      }
    }
  }
}
