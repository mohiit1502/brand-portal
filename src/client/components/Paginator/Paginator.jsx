import React, {Component} from 'react';
import PropTypes from 'prop-types';
import PaginationNav from "../custom-components/pagination/pagination-nav";
import CONSTANTS from "../../constants/constants";
import './Paginator.component.scss';

class Paginator extends Component {

  constructor(props) {
    super(props);
    this.changePageSize = this.changePageSize.bind(this);
    this.updatePage = this.updatePage.bind(this);
    this.state = {
      list: [],
      page: {
        offset: 0,
        size: 10,
        net: {
          size: 0,
          pagesCount: 0
        },
        sizeOptions: [5, 10, 15, 20, 30],
        list: []
      }
    }
  }

  componentDidMount() {
    if (this.props.records.length && this.state.page.offset >= 0 && this.state.page.size >= 0) {
      this.updatePage(this.state.page.offset, this.state.page.size);
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.records !== prevProps.records && this.state.page.offset >= 0 && this.state.page.size >= 0) {
      this.updatePage(this.state.page.offset, this.state.page.size);
    }
  }

  updatePage (offset, size) {
    const list = this.props.records;
    const pageState = {...this.state.page};
    pageState.size = size;
    pageState.net.size = list.length;
    pageState.net.pagesCount = Math.ceil(pageState.net.size / pageState.size);
    pageState.list = pageList;
    pageState.offset = offset >= pageState.net.pagesCount ? pageState.net.pagesCount - 1 : offset;
    const startIndex = pageState.offset * size;
    const endIndex = startIndex + size;
    const pageList = list.slice(startIndex, endIndex);

    if (pageState.offset < 0) return;
    const paginatedList = [...pageList];
    this.setState({list, page: pageState});
    this.props.updateListAndFilters(paginatedList);
  }

  changePageSize(size) {
    const page = {...this.state.page};
    page.size = size;
    this.setState({page}, () => this.updatePage(this.state.page.offset, size));
  }

  render() {
    const page = this.state.page;
    const records = this.props.records;
    const from = page.offset * page.size + 1;
    const to = page.offset * page.size + this.props.paginatedList.length;
    const total = records && records.length;
    let pageViewInfo = records && records.length ? <React.Fragment>
      Viewing <span className="count font-weight-bold" >{to >= from ? `${from} - ${to}` : 0}</span> of {total} {CONSTANTS[this.props.section].SECTION_TITLE_PLURAL}
    </React.Fragment> : ""

    const pageSizeSelector = records && records.length ?
      <React.Fragment>
        <span className="showing-content pr-2">Show</span>
        <button type="button" className="btn btn-sm count-toggle-btn dropdown-toggle px-4" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
          {page.size} {CONSTANTS[this.props.section].SECTION_TITLE_PLURAL} &nbsp;&nbsp;&nbsp;
        </button>
        <div className="dropdown-menu count-dropdown-menu">
          {page.sizeOptions.map(val => <a key={val} className="dropdown-item" onClick={() => {this.changePageSize(val);}}>
            {val} {CONSTANTS[this.props.section].SECTION_TITLE_PLURAL}</a>)}
        </div>
      </React.Fragment> : null;

    return (
      <div className="c-Paginator row table-manage-row h-10 px-4 align-items-center">
        <div className="col text-left">{pageViewInfo}</div>
        <div className="col text-center">
          <PaginationNav list={records ? records : []} page={this.state.page} updatePage={this.updatePage}/>
        </div>
        <div className="col text-right">{pageSizeSelector}</div>
      </div>
    );
  }
}

Paginator.propTypes = {
  createFilters: PropTypes.func,
  filteredList: PropTypes.array,
  paginatedList: PropTypes.array,
  records: PropTypes.array,
  section: PropTypes.string,
  updateListAndFilters: PropTypes.func
};

Paginator.defaultProps = {
  paginatedList: [],
  records: []
}

export default Paginator;
