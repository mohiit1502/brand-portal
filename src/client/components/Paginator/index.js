import Paginator from "./Paginator.jsx";

export default Paginator;
