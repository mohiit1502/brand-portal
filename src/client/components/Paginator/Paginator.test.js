import React from 'react';
import Paginator from './Paginator';

describe('Paginator', () => {
   it('renders without error', () => {

   });
});