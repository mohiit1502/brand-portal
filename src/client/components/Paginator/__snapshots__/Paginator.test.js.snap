// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator renders without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MockNextContext
    pathname="/help"
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/help",
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/help",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/help",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Paginator
          createFilters={[Function]}
          filteredList={Array []}
          paginatedList={Array []}
          records={Array []}
          section=""
          updateListAndFilters={[Function]}
        >
          <div
            className="c-Paginator row table-manage-row h-10 mb-4 align-items-center"
          >
            <div
              className="col text-left"
            />
            <div
              className="col text-center overflow-auto"
            >
              <Connect(PaginationNav)
                list={Array []}
                page={
                  Object {
                    "list": Array [],
                    "net": Object {
                      "pagesCount": 0,
                      "size": 0,
                    },
                    "offset": 0,
                    "size": 10,
                    "sizeOptions": Array [
                      5,
                      10,
                      15,
                      20,
                      30,
                    ],
                  }
                }
                updatePage={[Function]}
              >
                <PaginationNav
                  dispatch={[Function]}
                  list={Array []}
                  page={
                    Object {
                      "list": Array [],
                      "net": Object {
                        "pagesCount": 0,
                        "size": 0,
                      },
                      "offset": 0,
                      "size": 10,
                      "sizeOptions": Array [
                        5,
                        10,
                        15,
                        20,
                        30,
                      ],
                    }
                  }
                  updatePage={[Function]}
                />
              </Connect(PaginationNav)>
            </div>
            <div
              className="col text-right"
            />
          </div>
        </Paginator>
      </Router>
    </MemoryRouter>
  </MockNextContext>
</Provider>
`;
